modules/govuk/manifests/apps/asset_manager.pp

```diff
@@ -88,7 +88,40 @@
       location ~ /raw/(.*) {
         internal;
         alias /var/apps/asset-manager/uploads/assets/$1;
-      }',
+      }
+
+      location = /cloud-storage-proxy/ {
+        # Prevent requests to this location from outside nginx
+        internal;
+
+        # Extract download host and URL from upstream HTTP headers
+        set $cloud_storage_host $upstream_http_x_cloud_storage_host;
+        set $download_url $upstream_http_x_cloud_storage_url;
+
+        # The X-CLOUD-STORAGE-URL header contains a signed URL for the asset on
+        # S3. The signature of this URL is based in part on the request headers
+        # set in the asset manager rails app at the time the URL is generated.
```

Comment:
Very minor: I'd probably either hyphenate "asset-manager" and capitalise "Rails".